src/simulators/statevector/qv_avx2.cpp

```diff
@@ -1198,16 +1192,15 @@ Avx apply_diagonal_matrix_avx<double>(double* qv_data_,
     const auto until = base + (1UL << (batch + 1));
     for (auto i = base; i < until; i+=2) {
       auto tgt_qv_data = _mm256_load(reinterpret_cast<double*>(&(qv_data[i])));
-      auto input_data = _load_diagonal_input(in_vec, double_tls, i, qregs, qregs_size, q0_mask);
+      auto input_data = _load_diagonal_input(in_vec, double_tmp, i, qregs, qregs_size, q0_mask);
```

Comment:
In the OpenMP case here, how are we ensuring that there's no thread collision in the pointer?  I see that `double_tmp` is allocated space for `2 * omp_max_num_threads` complex doubles, but `_load_diagonal_input` seems to access this pointer at 0 offset.  Should the lambda get its thread number outside of the loop, and access `double_tmp` at an offset of `2 * omp_get_thread_num()` (or 0 if not OpenMP)?

Reply:
Thanks... This is my mistake. The reason why memory is allocated with `thradprivate` was to avoid conflict. Allocating two `double` or four `float` are necessary for each thread. `double_tmp` and `float_tmp` must be accessed via indexers.